twyla/kubedeploy/kube.py

```diff
@@ -76,11 +122,33 @@ def get_deployment(self):
                 raise e
 
 
-    def deploy(self, tag: str):
+    def get_remote_service(self):
+        try:
+            res = self.ext_v1_beta_client.read_namespaced_service(
+                name=self.deployment_name,
+                namespace=self.namespace)
+            return res
+        except kubernetes.client.rest.ApiException as e:
+            # Create a new service if no existing is found
+            if e.status == 404:
+                msg = "No service found for {}".format(self.deployment_name)
+                raise ServiceNotFoundException(msg) from None
+            else:
+                raise e
```

Comment:
The right pattern here is to just `raise`, as this will preserved the complete stack trace. If you do `raise e`, the stack trace will be overwritten with this point as the origin.